pytest_mitmproxy.py

```diff
@@ -1 +1,106 @@
 # -*- coding: utf-8 -*-
+import re
+import os
+import six
+import time
+import shutil
+import pytest
+from contextlib import contextmanager
+from tempfile import TemporaryFile
+if six.PY3:
+    from urllib.parse import urlparse
+    from subprocess import Popen, CalledProcessError, TimeoutExpired
+elif six.PY2:
+    from urlparse import urlparse
+    from subprocess import Popen, CalledProcessError
+
+# clean the dumps each run
+if os.path.isdir("traffic_dumps"):
+    shutil.rmtree("traffic_dumps")
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--mitmproxy-filter",
+        action="store",
+        default=None,
+        help="filter specification for mitmproxy"
+    )
+
+
+@contextmanager
+def new_proxy(name, scope, flt=None):
+    ''' Start a new mitmdump process and
+    return the address for clients to connect
+
+    The captured content will be written to
+    ./<dump_dir>/<scope>/<name>
+    '''
+    dump_dir = "traffic_dumps"
+
+    if six.PY3:
+        os.makedirs(os.path.join(dump_dir, scope), mode=0o755, exist_ok=True)
+    elif six.PY2:
+        try:
+            os.makedirs(os.path.join(dump_dir, scope), mode=0o755)
+        except OSError:
+            pass
+
+    dump_file = os.path.join(dump_dir, scope, name)
+
+    args = ["mitmdump", "--port=0", "--wfile={}".format(dump_file)]
+    args.append(flt) if flt else None
```

Comment:
Don't try and be clever like this:

```python
if ftl:
    args.append(flt)
```

Easy, straightfoward, no need to pause and realise you're relying on side-effects of an expression to replace control flow.